### PR TITLE
[WIP] Matrix norm results in wrong result

### PR DESCRIPTION
**Pull request by kritkaran94**
This PR addresses Issue #11413.

ccing @asmeurer, @jksuom.

### CODE REVIEW

**Comment by kritkaran94** [comment]
Output after this PR is shown below.

````
>>> v=symbols('v:3',real=True);v
(v0, v1, v2)
>>> V=Matrix(v);V
Matrix([
[v0],
[v1],
[v2]])
>>> U=V.normalized()
>>> U
Matrix([
[v0/sqrt(v0**2 + v1**2 + v2**2)],
[v1/sqrt(v0**2 + v1**2 + v2**2)],
[v2/sqrt(v0**2 + v1**2 + v2**2)]])
>>> U.norm()
sqrt(v0**2/(v0**2 + v1**2 + v2**2) + v1**2/(v0**2 + v1**2 + v2**2) + v2**2/(v0**2 + v1**2 + v2**2))
>>> _.simplify()
1
````

Any comments on the same? I feel there are some issues with regards to the test file.

File: sympy/functions/elementary/complexes.py [hunk 1]
```diff
@@ -467,7 +466,8 @@ def eval(cls, arg):
                     return Abs(base)**exponent
                 if base.is_positive == True:
```
Comment (jksuom) [inline sympy/functions/elementary/complexes.py]: A better form of this test is:
```
    if base.is_nonnegative:
```

File: sympy/functions/elementary/complexes.py [hunk 2]
```diff
@@ -467,7 +466,8 @@ def eval(cls, arg):
...
                     return base**re(exponent)
-                return (-base)**re(exponent)*exp(-S.Pi*im(exponent))
+                if base.is_negative:
+                    return (-base)**re(exponent)*exp(-S.Pi*im(exponent))
```
Comment (jksuom) [inline sympy/functions/elementary/complexes.py]: After this line we could have `return` alone to take care of the remaining cases where neither `base.is_nonnegative` nor `base.is_negative` is satisfied.

File: sympy/functions/elementary/complexes.py (hunk 1 above)
Comment (kritkaran94) [inline sympy/functions/elementary/complexes.py]: I will make both the changes and get back to you.

File: sympy/functions/elementary/complexes.py (hunk 2 above)
Comment (kritkaran94) [inline sympy/functions/elementary/complexes.py]: @jksuom Travis points build to fail w.r.t to the inequalities file, but I din't change that file?

**Comment by jksuom** [comment]
If `bin/test inequalities` continues to fail after these changes in `Abs` (as it will probably do), the tests in `test_inequalities` should be fixed.

**Comment by kritkaran94** [comment]
Okay, Will try to work on your feedback as soon as possible.

**Comment by kritkaran94** [comment]
@jksuom I have made the changes.

File: sympy/functions/elementary/complexes.py [hunk 3]
```diff
@@ -465,9 +464,11 @@ def eval(cls, arg):
                     if base.func is cls and exponent is S.NegativeOne:
                         return arg
                     return Abs(base)**exponent
-                if base.is_positive == True:
-                    return base**re(exponent)
-                return (-base)**re(exponent)*exp(-S.Pi*im(exponent))
+                    if base.is_nonnegative:
```
Comment (jksuom) [inline sympy/functions/elementary/complexes.py]: It seems that the indentation was changed. I should be the same as before.

Comment (kritkaran94) [inline sympy/functions/elementary/complexes.py]: Okay, right away.

Comment (jksuom) [inline sympy/functions/elementary/complexes.py]: This line (and the next) as well.

Comment (kritkaran94) [inline sympy/functions/elementary/complexes.py]: Line 471. Right ? (the one with ``return``)

Comment (jksuom) [inline sympy/functions/elementary/complexes.py]: These two lines:
```
                   if base.is_nonnegative:
                        return base**re(exponent)
```
They should be on the same level as lines 469 and 470

Comment (kritkaran94) [inline sympy/functions/elementary/complexes.py]: Done that, in reference to the original file.

````
if base.is_nonnegative:
         return base**re(exponent)
if base.is_negative:
         return (-base)**re(exponent)*exp(-S.Pi*im(exponent))
````

[This line] (https://github.com/sympy/sympy/blob/master/sympy/functions/elementary/complexes.py#L468l) is exactly below ``if exponent.is_integer:``

**Comment by kritkaran94** [comment]
@jksuom I have made all the changes. I Apologize in advance if I misunderstood any of your comments. 

However, ``bin/test inequalities`` continued to fail after these changes in ``Abs``, the tests in ``test_inequalities`` didn't pass.

**Comment by jksuom** [comment]
The code looks good. There is an error in sympy/solvers/tests/test_inequalities.py:test_issue_10198:
```
    assert reduce_inequalities(abs(1/sqrt(x)) - 1, x) == Eq(x, -1)
```

 The current implementation will mistakenly omit `abs` and change the sign of `x`:
```
>>> x = symbols('x', real=True)
>>> abs(1/sqrt(x))
1/sqrt(-x)
```
It then solves the equation `1/sqrt(-x) - 1 = 0` obtaining the incorrect result `x = -1`.
The assertion should be replaced with a line similar to the last one with `NotImplementedError` instead of `ValueError`:
```
    raises(NotImplementedError, lambda: reduce_inequalities(abs(1/sqrt(x)) - 1, x))
```

**Comment by kritkaran94** [comment]
@jksuom Thanks for that. 

In ``sympy/solvers/tests/test_inequalities.py`` file  now we have the following line :

Before this PR [Line 326](https://github.com/sympy/sympy/pull/11417/files#diff-b3fe41ada1b7b645fe6a140b3ae5da32L326)

````
assert reduce_inequalities(abs(1/sqrt(x)) - 1, x) == Eq(x, -1)
````
After this PR (Line 326)
````
raises(NotImplementedError, lambda: reduce_inequalities(abs(1/sqrt(x)) - 1, x))
````

**Comment by kritkaran94** [comment]
@jksuom All the tests have passed. Is this PR ready for merge ?

**Comment by jksuom** [comment]
Yes, I think so. Thanks.

**Comment by kritkaran94** [comment]
Thanks a lot for your guidance.

**Pull request merged**